llms/mlx_lm/utils.py

```diff
@@ -550,6 +550,47 @@ def quantize_model(
     return quantized_weights, quantized_config
 
 
+def update_config(
+    config: dict,
+    upload_repo: Optional[str] = None,
+    config_path: Optional[Union[str, Path]] = None,
+    indent: int = 4,
+    **kwargs,
+) -> dict:
+    """Update the model configuration and save config to the ``config_path`` (if provided).
```

Comment:
Should we make the config_path mandatory and always save the configuration? We could even name it "save_config". I don't see another use case where we would only update the configuration without saving it.

Reply:
It's a good point. However, there could be scenarios, when you could have a use-case of updating the config dynamically on the fly, and may want to skip I/O interactions until the end -- when you just write to the disc just once.

I thought of leaving that option out there so that the same function can potentially fill in both scenarios --  dynamic in-memory config update vs. immediate file (config.json) update. But I am open to suggestions on this one.

About calling the variable `save_config`, I have a little reservation though, as `save_config` sounds like a verb and not a noun as `config_path`. Also, you could be reading config from an existing `config.json` file and then updating it later (such as sorting, adding more key-value pairs, etc.). Thus, the `config_path` refers to a path where either a new config.json file will be created or an existing config will be updated.

Reply:
It would be clear to split this function into two: one for updating the config and another for saving the config. Then you can compose the functions for different use cases. what do you think?

Reply:
Yes, maintaining single responsibility would be preferable. I will refactor it into two functions then:

- `update_config()`
- `save_config()`

What do you think about it?